Sources/MapboxCoreNavigation/Feedback/FeedbackMetadata.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+import CoreLocation
+import MapboxNavigationNative
+@_implementationOnly import MapboxNavigationNative_Private
+
+/// :nodoc:
+@_spi(MapboxInternal)
+public struct FeedbackMetadata {
+    private let userFeedbackHandle: NativeUserFeedbackHandle?
+    private var calculatedUserFeedbackMetadata: UserFeedbackMetadata? = nil
+
+    var userFeedbackMetadata: UserFeedbackMetadata? {
+        calculatedUserFeedbackMetadata ?? userFeedbackHandle?.getMetadata()
+    }
+
+    public let screenshot: String?
+    public internal(set) var location: CLLocation?
```

Comment:
I revisited this after comparing it to Android and decided to create a new `UserFeedback` model.

----

This location is the equivalent for Android [location](https://github.com/mapbox/mapbox-navigation-android/blob/687f6ba31514c86aaee3fc3c88d8818ee602b237/libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/telemetry/UserFeedback.kt#L19) in `UserFeedback`.

Since NN is responsible for actual event sending after migration to the new Telemetry we can get location only from NN. In this case, we get it in the callback. This potentially can be used in Copilot.